azure-pipelines-docs-integration.yml

```diff
@@ -52,13 +52,27 @@ stages:
                 tests/integration/fixtures/**
                 tests/test_sets/**
 
+          - bash: |
+              CHANGED_FILES=$(git diff HEAD origin/develop --name-only)
+              DEPENDENCIES=$(python scripts/trace_docs_deps.py docs)
+              printf '%s\n' "[CHANGED FILES]" "${CHANGED_FILES[@]}" "" "[DEPENDENCIES]" "${DEPENDENCIES[@]}" ""
+
+              echo "Checking for file changes from 'great_expectations/' that impact 'docs/'..."
+              for FILE in $DEPENDENCIES; do
+                if [[ ${CHANGED_FILES[@]} =~ $FILE ]]; then
+                    echo "Found change in local dependency:" $FILE
+                    echo "##vso[task.setvariable variable=DocsDependenciesChanged;isOutput=true]true"
+                    break
+                fi
+              done
+            name: CheckDocsDependenciesChanges
 
   - stage: docusaurus_tests
     pool:
       vmImage: 'ubuntu-latest'
     jobs:
     - job: test_docs
-      condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckChanges.GEChanged'], true), eq(stageDependencies.scope_check.changes.outputs['CheckChanges.DocsChanged'], true), eq(variables.isMain, true))
+      condition: or(eq(stageDependencies.scope_check.changes.outputs['CheckDocsDependenciesChanges.DocsDependenciesChanged'], true), eq(stageDependencies.scope_check.changes.outputs['CheckDocsChanges.DocsChanged'], true), eq(variables.isMain, true))
```

Comment:
This previously used `CheckChanges.GEChanged`, which is a condition in our primary Azure YAML. I think looking at every single change in GE is far too costly for our CI/CD.

In order for our docs pipeline to run, we must fulfill one of two conditions:
  - Something in our `docs/` changed.
  - A local dependency used within `docs/` has changed.

Reply:
Brilliant!